OpenMW specific changes

elsid opened the pull request:
* Add a new port openmw-osg to build package for https://github.com/OpenMW/osg. It has just basic support for options OpenMW needs like specific set of plugins.
* Add support for dynamic linking to MyGUI. vcpkg support only static libraries for it due to issues in the platform libraries. Make those libraries static and make main one to support static and dynamic linking.
* Add features to MyGUI to make the build compatible with OpenMW. Add `MYGUI_DONT_USE_OBSOLETE` macro support and ability to disable unused options.

elsid committed:
Add openmw-osg: osg fork for openmw

elsid committed:
Add plugins and obsolete features to mygui

To be able to build it without plugins and obsolete functions.

elsid committed:
Support dynamic linking for mygui

AnyOldName3 commented:
I can't write comments on the files for some reason, but I think we want to add an option to build MyGUI with no rendersystem, which is what we normally do, then we can make it so MyGUI can only be dynamic without a rendersystem, and drop the patches to make the rendersystems we don't use work.

I'm also not sure it's wise to put OSG in UTF-8 mode. We know from experience that it's harder than it looks to interact with the Windows filesystem API with UTF-8, and we don't know if anyone's been using that code, so it could be buggy, whereas we know we're not feeding OSG anything it chokes on in the default mode as we've been using it for years.

elsid commented:
I've removed UTF-8 for osg and added a feature to build mygui with export render system but I also had to make a patch. Looking at the CMake files, the library is not ready for this. There is [mygui_set_platform_name](https://github.com/MyGUI/mygui/blob/17d66baea79b415fec94412d2bb94d847f81696c/CMake/Utils/MyGUIConfigTargets.cmake#L27-L43) which sets `MYGUI_PLATFORM_NAME` depending on the render system and then used all over the place to define paths, targets etc. In the case of `MYGUI_RENDERSYSTEM=2` it's not set so many things just don't work. The most obvious one is [add_subdirectory](https://github.com/MyGUI/mygui/blob/17d66baea79b415fec94412d2bb94d847f81696c/Platforms/CMakeLists.txt#L3) with such path. I wonder how this works in other builds.

AnyOldName3 commented:
Looks like I was a little muddled, then - we use the dummy platform, which is still a platform, rather than there being a way to build it without a platform.
